File: ai-community-backend/src/main/java/com/aicommunity/entity/Message.java
package com.aicommunity.entity;

import lombok.Data;
import java.io.Serializable;
import java.time.LocalDateTime;

/**
 * 消息实体类
 *
 * @author AI Community Team
 */
@Data
public class Message implements Serializable {

    private static final long serialVersionUID = 1L;

    /**
     * 消息ID
     */
    private Long id;

    /**
     * 接收用户ID
     */
    private Long userId;

    /**
     * 消息类型：activity_registration-活动报名，post_comment-帖子评论，comment_reply-评论回复，post_like-帖子点赞，award_notification-获奖通知
     */
    private String type;

    /**
     * 消息标题
     */
    private String title;

    /**
     * 消息内容
     */
    private String content;

    /**
     * 跳转链接
     */
    private String link;

    /**
     * 发送用户ID
     */
    private Long fromUserId;

    /**
     * 是否已读：0-未读，1-已读
     */
    private Boolean read;

    /**
     * 创建时间
     */
    private LocalDateTime createTime;
}

File: frontend/src/api/message.ts
// 消息相关接口
import request from './request'

// 消息类型
export enum MessageType {
  ACTIVITY_REGISTRATION = 'activity_registration', // 活动报名
  POST_COMMENT = 'post_comment', // 帖子评论
  COMMENT_REPLY = 'comment_reply', // 评论回复
  POST_LIKE = 'post_like', // 帖子点赞
  AWARD_NOTIFICATION = 'award_notification' // 奖项通知
}

// 消息接口
export interface Message {
  id: number
  userId: number
  type: string
  title: string
  content: string
  link?: string
  fromUserId?: number
  fromUserName?: string
  read: boolean
  createTime: string | Date
}

// 分页响应
export interface PageResult<T> {
  list: T[]
  total: number
  page: number
  pageSize: number
}

// 未读消息数量响应
export interface MessageUnreadCountResponse {
  count: number
}

// 获取消息列表
export const getMessages = async (params?: {
  type?: string
  page?: number
  pageSize?: number
}): Promise<PageResult<Message>> => {
  return await request.get<PageResult<Message>>('/messages', { params })
}

// 标记消息为已读
export const markMessageAsRead = async (id: number): Promise<void> => {
  await request.put(`/messages/${id}/read`)
}

// 全部标记为已读
export const markAllMessagesAsRead = async (): Promise<void> => {
  await request.put('/messages/read-all')
}

// 删除消息
export const deleteMessage = async (id: number): Promise<void> => {
  await request.delete(`/messages/${id}`)
}

// 获取未读消息数量
export const getUnreadMessageCount = async (): Promise<MessageUnreadCountResponse> => {
  return await request.get<MessageUnreadCountResponse>('/messages/unread-count')
}
